### PR TITLE
Stshapir/vz 4021

### PR DESCRIPTION
**Pull request by stevshap**
# Description

This pull request adds the framework to push metrics from test suites using the VzDescribe wrappers. Only ci/kind tests currently have Prometheus credentials enabled; will be resolved in future changes. Also resolves error of invalid metrics name failure

Fixes VZ-9999

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: pkg/test/framework/ginkgo_wrapper.go [hunk 1]
```diff
@@ -69,8 +70,23 @@ func VzAfterEach(body interface{}) bool {
...
+		startTime := time.Now()
+
+		pkg.Log(pkg.Info, fmt.Sprintf("Describe block %q started - placeholder for making API call to emit test related metric(s)", ginkgo.CurrentSpecReport().LeafNodeText))
+		reflect.ValueOf(body).Call([]reflect.Value{})
+		pkg.Log(pkg.Info, fmt.Sprintf("Describe block %q ended - placeholder for making API call to emit test related metric(s)", ginkgo.CurrentSpecReport().LeafNodeText))
+
+		endTime := time.Now()
+		durationMillis := float64(endTime.Sub(startTime) / time.Millisecond)
+
+		if EmitGauge(text, "duration", durationMillis) != nil {
```
Comment (markxnelson) [inline pkg/test/framework/ginkgo_wrapper.go:83]: `text` might not be the best choice here?  are there any implication for it having spaces in it - how does prometheus handle that?  
i wonder if there is a better way of coming up with a good metric name - maybe we want the hierarchy, or the file at least?

Comment (markxnelson) [reply pkg/test/framework/ginkgo_wrapper.go:83]: i see you have a func makeMetricName, but you don't seem to use it here? - oh i see you call it down further  - ok

**Review by markxnelson — APPROVED** [review]

File: pkg/test/framework/ginkgo_wrapper.go [hunk 2]
```diff
@@ -6,8 +6,9 @@ package framework
 import (
 	"fmt"
 	"reflect"
+	"time"
 
-	ginkgo "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/ginkgo/v2"
```
Comment (bobdonat) [inline pkg/test/framework/ginkgo_wrapper.go:11]: looks like the PR was already merged.
my understanding is we decided to use a dot import for ginkgo
should be 
. "github.com/onsi/ginkgo/v2"

Comment (stevshap) [reply pkg/test/framework/ginkgo_wrapper.go:11]: ack

**Pull request merged**